File: src/services/enums.ts
export type OtherSpecify = 'Other';

export const patientSexValues = ['Male', 'Female'] as const;
export type PatientSex = typeof patientSexValues[number];

export const patientGenderValues = ['Male', 'Female'] as const;
export type PatientGender = typeof patientGenderValues[number];

export const patientPronounValues = ['He/Him', 'She/Her', 'They/Them'] as const;
export type PatientPronoun = typeof patientPronounValues[number];

export const patientRaceEthnicityValues = ['White', 'Black'] as const;
export type PatientRaceEthnicity = typeof patientRaceEthnicityValues[number];

export const clinicCodeValues = [
    'Breast',
    'Endocrine',
    'GI',
    'GI – Pancreatic',
    'GU',
    'GYN',
    'HEME',
    'HEME – Sickle Cell',
    'HNL',
    'Immunotherapy',
    'Melanoma/Renal',
    'Neuro',
    'NW Hospital',
    'Sarcoma',
    'Transplant – Auto',
    'Transplant – Allo',
    'Transplant – CAR-T',
    'Transplant – LTFU',
    'Transplant – TTC',
    'Other',
] as const;
export type ClinicCode = typeof clinicCodeValues[number];

export type AllClinicCode = 'All Clinics';

export const depressionTreatmentStatusValues = [
    'CoCM',
    'CoCM Relapse Prevention',
    'Discharged',
    'Pending',
    'Other',
] as const;
export type DepressionTreatmentStatus = typeof depressionTreatmentStatusValues[number];

export const followupScheduleValues = ['1-week follow-up', '2-week follow-up', '4-week follow-up'] as const;
export type FollowupSchedule = typeof followupScheduleValues[number];

export const discussionFlagValues = ['Flag as safety risk', 'Flag for discussion'] as const;
export type DiscussionFlag = typeof discussionFlagValues[number];

export const cancerTreatmentRegimenValues = [
    'Surgery',
    'Chemotherapy',
    'Radiation',
    'Stem Cell Transplant',
    'Immunotherapy',
    'CAR-T',
    'Endocrine',
    'Surveillance',
    'Other',
] as const;
export type CancerTreatmentRegimen = typeof cancerTreatmentRegimenValues[number] | OtherSpecify;

export const referralValues = [
    'None',
    'Psychiatry',
    'Psychology',
    'Pt Navigation',
    'Integrative Medicine',
    'Spiritual Care',
    'Palliative Care',
] as const;
export type Referral = typeof referralValues[number] | OtherSpecify;

export const sessionTypeValues = ['In person at clinic', 'Telehealth', 'Phone', 'Group', 'Home'] as const;
export type SessionType = typeof sessionTypeValues[number] | OtherSpecify;

export const treatmentPlanValues = [
    'Maintain current treatment',
    'Adjust treatment plan',
    'Monitor only',
    'No further follow-up',
    'Refer to community',
] as const;
export type TreatmentPlan = typeof treatmentPlanValues[number];

export const treatmentChangeValues = ['None', 'Medication', 'Counseling'] as const;
export type TreatmentChange = typeof treatmentChangeValues[number] | OtherSpecify;

export const behavioralActivationChecklistValues = [
    'Review of the BA model',
    'Values and goals assessment',
    'Activity scheduling',
    'Mood and activity monitoring',
    'Relaxation',
    'Contingency management',
    'Managing avoidance behaviors',
    'Problem-solving',
] as const;
export type BehavioralActivationChecklistItem = typeof behavioralActivationChecklistValues[number];

export const assessmentFrequencyValues = ['Daily', 'Once a week', 'Every 2 weeks', 'Monthly'] as const;
export type AssessmentFrequency = typeof assessmentFrequencyValues[number];

export const phq9ItemValues = [
    'Interest',
    'Feeling',
    'Sleep',
    'Tired',
    'Apetite',
    'Failure',
    'Concentrating',
    'Slowness',
    'Suicide',
] as const;
export type PHQ9Item = typeof phq9ItemValues[number];

export const gad7ItemValues = [
    'Anxious',
    'Constant worrying',
    'Worrying too much',
    'Trouble relaxing',
    'Restless',
    'Irritable',
    'Afraid',
] as const;
export type GAD7Item = typeof gad7ItemValues[number];

File: server_flask/enums.py
from enum import Enum


class ClinicCode(Enum):
    Breast = "Breast"
    Endocrine = "Endocrine"
    GI = "GI"
    GIPancreatic = "GI – Pancreatic"
    GU = "GU"
    GYN = "GYN"
    HEME = "HEME"
    HEMESickleCell = "HEME – Sickle Cell"
    HNL = "HNL"
    Immunotherapy = "Immunotherapy"
    MelanomaRenal = "Melanoma/Renal"
    Neuro = "Neuro"
    NWHospital = "NW Hospital"
    Sarcoma = "Sarcoma"
    TransplantAuto = "Transplant – Auto"
    TransplantAllo = "Transplant – Allo"
    TransplantCART = "Transplant – CAR-T"
    TransplantLTFU = "Transplant – LTFU"
    TransplantTTC = "Transplant – TTC"
    Other = "Other"


class PatientSex(Enum):
    Male = "Male"
    Female = "Female"


class PatientGender(Enum):
    Male = "Male"
    Female = "Female"


class PatientRaceEthnicity(Enum):
    White = "White"
    Black = "Black"


class PatientPronoun(Enum):
    HeHim = "He/Him"
    SheHer = "She/Her"
    TheyThem = "They/Them"


class DepressionTreatmentStatus(Enum):
    CoCM = "CoCM"
    CoCMRelapse = "CoCM Relapse Prevention"
    Discharged = "Discharged"
    Pending = "Pending"
    Other = "Other"


class FollowupSchedule(Enum):
    OneWeek = "1-week follow-up"
    TwoWeeks = "2-week follow-up"
    FourWeeks = "4-week follow-up"


class DiscussionFlag(Enum):
    SafetyRisk = "Flag as safety risk"
    Discussion = "Flag for discussion"


class CancerTreatmentRegimen(Enum):
    Surgery = "Surgery"
    Chemotherapy = "Chemotherapy"
    Radiation = "Radiation"
    StemCellTransplant = "Stem Cell Transplant"
    Immunotherapy = "Immunotherapy"
    CART = "CAR-T"
    Endocrine = "Endocrine"
    Surveillance = "Surveillance"
    Other = "Other"


class Referral(Enum):
    Psychiatry = "Psychiatry"
    Psychology = "Psychology"
    PtNavigation = "Pt Navigation"
    IntegrativeMedicine = "Integrative Medicine"
    SpiritualCare = "Spiritual Care"
    PalliativeCare = "Palliative Care"
    Other = "Other"


class ReferralStatus(Enum):
    NotReferral = "Not Referred"
    Pending = "Pending"
    Active = "Active"
    Completed = "Completed"


class SessionType(Enum):
    InPerson = "In person"
    Telehealth = "Telehealth"
    Phone = "Phone"
    Group = "Group"
    Home = "Home"


class TreatmentPlan(Enum):
    MaintainPlan = "Maintain current treatment"
    AdjustPlan = "Adjust treatment plan"
    MonitorOnly = "Monitor only"
    NoFollowup = "No further follow-up"
    ReferToCommunity = "Refer to community"


class TreatmentChange(Enum):
    NoChange = "None"
    Medication = "Medication"
    Counseling = "Counseling"


class BehavioralActivationChecklist(Enum):
    ReviewModel = "Review of the BA model"
    ValuesGoals = "Values and goals assessment"
    ActivityScheduling = "Activity scheduling"
    MoodActivityMonitoring = "Mood and activity monitoring"
    Relaxation = "Relaxation"
    ContingencyManagement = "Contingency management"
    ManagingAvoidance = "Managing avoidance behaviors"
    ProblemSolving = "Problem-solving"


class BehavioralStrategyChecklist(Enum):
    BehavioralActivation = "Behavioral Activation"
    MotivationalInterviewing = "Motivational Interviewing"
    ProblemSolvingTherapy = "Problem Solving Therapy"
    CognitiveTherapy = "Cognitive Therapy"
    MindfulnessStrategies = "Mindfulness Strategies"
    SupportiveTherapy = "Supportive Therapy"
    Other = "Other"


class AssessmentType(Enum):
    PHQ9 = "PHQ-9"
    GAD7 = "GAD-7"
    MoodLogging = "Mood Logging"
    Medication = "Medication Tracking"


class AssessmentFrequency(Enum):
    Daily = "Daily"
    Weekly = "Once a week"
    Biweekly = "Every 2 weeks"
    Monthly = "Monthly"

class DayOfWeek(Enum):
    Monday = "Monday"
    Tuesday = "Tuesday"
    Wednesday = "Wednesday"
    Thursday = "Thursday"
    Friday = "Friday"
    Saturday = "Saturday"
    Sunday = "Sunday"
